src/edgepi/pwm/edgepi_pwm.py

```diff
@@ -0,0 +1,126 @@
+
+"""PWM Module"""
+
+import logging
+
+from edgepi.peripherals.pwm import PwmDevice
+
+from edgepi.gpio.gpio_constants import GpioPins
+from edgepi.gpio.edgepi_gpio import EdgePiGPIO
+from edgepi.pwm.pwm_constants import PWMCh, Polarity
+
+class EdgePiPWM():
+    """PWM module to provide PWM signal"""
+    __pwm_pin_to_channel = {GpioPins.PWM1 : PWMCh.PWM_1,
+                            GpioPins.PWM2 : PWMCh.PWM_2}
+
+    """handling PWM output"""
+    def __init__(self, pwm_num: GpioPins):
+        self.log = logging.getLogger(__name__)
+        # Control internal mux to enable/disable PWM
+        self.pwm_num = pwm_num
+        self.gpio = EdgePiGPIO()
+        self.pwm = PwmDevice(channel=self.__pwm_pin_to_channel[pwm_num].value.channel,
+                         chip=self.__pwm_pin_to_channel[pwm_num].value.chip)
+        self.channel = self.__pwm_pin_to_channel[pwm_num].value.channel
+        self.chip = self.__pwm_pin_to_channel[pwm_num].value.chip
+        self.pwm.open_pwm()
+
+    def set_frequency(self, frequency: int):
```

Comment:
is there a limit for frequency that we need to check?

Reply:
> is there a limit for frequency that we need to check?

good point, I've checked and saw signal deteriorates after 5KHz

Reply:
7da5f04c6f6cd9b18232ec5dceaa623149c0b17f